metadetect/lsst_measure_scarlet.py

```diff
@@ -0,0 +1,992 @@
+    band_sources = sources[subtractor.filters[0]]
+
+    # this gets everything that is a parent
+    parents = band_sources.getChildren(0)
+
+    for meas_parent_record in parents:
+        LOG.debug('-'*70)
+        LOG.debug('parent id: %d', meas_parent_record.getId())
+
+        children = band_sources.getChildren(meas_parent_record.getId())
+        nchild = len(children)
+        LOG.debug(
+            'processing %d %s',
+            nchild, 'children' if nchild > 1 else 'child'
+        )
+        results += _process_sources(
+            subtractor=subtractor, sources=children, stamp_size=stamp_size,
+            detexp=detexp,
+            wcs=wcs, fitter=fitter, show=show,
+        )
+
+    if len(results) > 0:
+        results = np.hstack(results)
+    else:
+        results = None
+
+    return results
+
+
+def _process_sources(
+    subtractor, sources, stamp_size, wcs, fitter, detexp, show=False,
+):
+    results = []
+    for source in sources:
+        res = _process_source(
+            subtractor=subtractor, source=source, stamp_size=stamp_size,
+            detexp=detexp,
+            wcs=wcs, fitter=fitter, show=show,
+        )
+        results.append(res)
+
+    return results
+
+
+def _process_source(
+    subtractor, source, stamp_size, wcs, fitter, detexp, show=False,
+):
+    source_id = source.getId()
+
+    ormask = get_ormask(source=source, exposure=detexp)
+    exp_bbox = detexp.getBBox()
+
+    res = None
+    with subtractor.add_source(source_id):
+
+        if show:
+            vis.show_mbexp(
+                subtractor.mbexp, mess=f'source {source_id} added'
+            )
+
+        try:
+            stamp_mbexp = subtractor.get_stamp(
+                source_id, stamp_size=stamp_size,
+            )
+
+            if show:
+                ostamp_mbexp = subtractor.get_stamp(
+                    source_id, stamp_size=stamp_size, type='original',
+                )
+                model_mbexp = subtractor.get_stamp(
+                    source_id, stamp_size=stamp_size, type='model',
+                )
+                vis.show_three_mbexp(
+                    [ostamp_mbexp, stamp_mbexp, model_mbexp],
+                    labels=['original', 'deblended', 'model']
+                )
+
+            # TODO make codes work with MultibandExposure rather than on a
+            # coadd of the bands
+
+            coadded_stamp_exp = util.coadd_exposures(stamp_mbexp.singles)
+            obs = _extract_obs(wcs=wcs, subim=coadded_stamp_exp, source=source)
+            if obs is None:
+                LOG.info('skipping object with all zero weights')
+                ores = {'flags': procflags.ZERO_WEIGHTS}
+                pres = {'flags': procflags.NO_ATTEMPT}
+                box_size = -1
+            else:
+                pres = _measure_one(obs=obs.psf, fitter=fitter)
+                ores = _measure_one(obs=obs, fitter=fitter)
+                box_size = obs.image.shape[0]
```

Comment:
Yes, but first need to write something that will do a robust centroid.

Reply:
For context:  the DM centroid code won't run on these stamps because it demands some huge radius around the object to do a centroid, like bigger than our stamps.  Easiest I think to just write a quick centroider.

Reply:
I'll be doing that in a separate PR